ganga/GangaCore/GPIDev/Lib/Registry/JobRegistry.py

```diff
@@ -84,6 +87,71 @@ def startup(self):
         self.flush_thread = RegistryFlusher(self, 'JobRegistryFlusher')
         self.flush_thread.start()
 
+    def check(self):
+        """
+            This code checks for jobs which are in the submitting state
+            If a job is 'submitting' an call to force_status('failed') is made
+            This should be enough to fix some future issues with monitoring """
+
+        def _shouldAutoKill(this_job):
+            """ This checks to see if the job is in a transistional state
+                By construction jobs in this state on disk need to be failed
+            """
+            return this_job.status in ["submitting", "completing"]
```

Comment:
Do we really want to include `completing` here. If ganga crashed whilst the job is finalising then surely we can push back to `submitted` and it will do the finalising again?

Reply:
If the job gets stuck in `completing` we don't want to start spinning this round in circles trying to manage the job. It's much better to mark it as failed and have the user manually intervene if they know the job ran successfully through the DIRAC web-UI.
This could be made configurable but I think there's a risk of having a job running around in a loop getting stuck not being parsed due to some external problem to ganga.

Reply:
I've spent some time with a user recently and we've confirmed that this seems to be the best approach as it allows ganga to get back into a helpful state whilst signalling to the user there is a potential problem with the subjob.